lib/file-source/src/file_watcher/mod.rs

```diff
@@ -292,9 +292,12 @@ impl FileWatcher {
 
 fn is_gzipped(r: &mut io::BufReader<fs::File>) -> io::Result<bool> {
     let header_bytes = r.fill_buf()?;
-    // WARN: The paired `BufReader::consume` is not called intentionally. If we
-    // do we'll chop a decent part of the potential gzip stream off.
-    Ok(header_bytes.starts_with(GZIP_MAGIC))
+    if header_bytes.starts_with(GZIP_MAGIC) {
+        r.consume(GZIP_MAGIC.len()); // To avoid interfering with the next read
+        Ok(true)
+    } else {
+        Ok(false)
+    }
```

Comment:
I think this is wrong. `is_gzipped` should just be checking if the stream is gzip and not consuming any bytes, as the comment says.